client_cli/src/main.rs

```diff
@@ -117,21 +117,29 @@ pub trait RunArgs {
     ///
     /// # Errors
     /// if inner command errors
-    fn run(self, cfg: &ClientConfiguration) -> Result<Box<dyn Serialize>>;
+    fn run<F: FnMut(&dyn Serialize) -> Result<()>>(
+        self,
+        cfg: &ClientConfiguration,
+        callback: F,
+    ) -> Result<()>;
 }
 
 macro_rules! match_all {
-    (($self:ident, $cfg:ident), { $($variants:path),* $(,)?}) => {
+    (($self:ident, $cfg:ident, $callback:ident), { $($variants:path),* $(,)?}) => {
         match $self {
-            $($variants(variant) => RunArgs::run(variant, $cfg),)*
+            $($variants(variant) => RunArgs::run(variant, $cfg, $callback),)*
         }
     };
 }
 
 impl RunArgs for Subcommand {
-    fn run(self, cfg: &ClientConfiguration) -> Result<Box<dyn Serialize>> {
+    fn run<F: FnMut(&dyn Serialize) -> Result<()>>(
```

Comment:
`FnMut(&dyn Serialize) -> Result<()>` is repeated over and over and over again. Could you use an alias for that?

Reply:
I also don't understand why it should be `FnMut` and not `Fn`?